benchto-driver/src/main/java/com/teradata/benchto/driver/loader/BenchmarkLoader.java

```diff
@@ -139,32 +137,65 @@
         }
     }
 
-    private List<Path> findBenchmarkFiles()
+    private List<Resource> findBenchmarkFiles(ResourcePatternResolver patternResolver)
             throws IOException
     {
-        LOGGER.info("Searching for benchmarks in classpath ...");
+        String benchmarksDir = properties.getBenchmarksDir();
+        List<String> activeBenchmarkPatterns = activeBenchmarkPatterns();
 
-        List<Path> benchmarkFiles = Files
-                .walk(properties.benchmarksFilesPath())
-                .filter(file -> isRegularFile(file) && file.toString().endsWith(BENCHMARK_FILE_SUFFIX))
+        LOGGER.info("Searching for {} benchmarks in  {} ...", activeBenchmarkPatterns, benchmarksDir);
+
+        List<Resource> benchmarkFiles = activeBenchmarkPatterns.stream()
+                .flatMap(pattern -> getResources(patternResolver, benchmarksDir + "/" + pattern))
+                .peek(resource -> LOGGER.info("Benchmark found: {}", resource))
                 .collect(toList());
-        benchmarkFiles.stream().forEach((path) -> LOGGER.info("Benchmark found: {}", path.toString()));
 
         return benchmarkFiles;
     }
 
-    private List<Benchmark> loadBenchmarks(String sequenceId, List<Path> benchmarkFiles)
+    private ResourcePatternResolver getResourcePatternResolver()
+    {
+        FileSystemResourceLoader resourceLoader = new FileSystemResourceLoader()
```

Comment:
Can you please extract this as separate named class and then please add javadoc which explains why do we need to override it?